test/compoundPricer.test.ts

```diff
@@ -0,0 +1,133 @@
+    weth = await MockERC20.new('WETH', 'WETH', 18)
+    cETH = await MockCToken.new('CETH', 'CETH')
+    //
+    chainlinkPricer = await MockPricer.new(weth.address, oracle.address)
+  })
+
+  describe('constructor', () => {
+    it('should deploy the contract successfully', async () => {
+      compoundPricer = await CompoundPricer.new(cETH.address, weth.address, chainlinkPricer.address, oracle.address)
+    })
+    it('should revert if initializing with cToken = 0', async () => {
+      await expectRevert(
+        CompoundPricer.new(ZERO_ADDR, weth.address, chainlinkPricer.address, oracle.address),
+        'CompoundPricer: cToken address can not be 0',
+      )
+    })
+    it('should revert if initializing with underlying = 0 address', async () => {
+      await expectRevert(
+        CompoundPricer.new(cETH.address, ZERO_ADDR, chainlinkPricer.address, oracle.address),
+        'CompoundPricer: underlying address can not be 0',
+      )
+    })
+    it('should revert if initializing with underlying pricer = 0', async () => {
+      await expectRevert(
+        CompoundPricer.new(cETH.address, weth.address, ZERO_ADDR, oracle.address),
+        'CompoundPricer: underlying pricer address can not be 0',
+      )
+    })
+    it('should revert if initializing with oracle = 0 address', async () => {
+      await expectRevert(
+        CompoundPricer.new(cETH.address, weth.address, chainlinkPricer.address, ZERO_ADDR),
+        'CompoundPricer: oracle address can not be 0',
+      )
+    })
+  })
+
+  describe('getPrice for cETH', () => {
+    const ethPrice = createScaledNumber(470)
+    const exchangeRate = new BigNumber('200192735438752381581313918')
+    before('mock data in chainlink pricer and cToken', async () => {
+      await chainlinkPricer.setPrice(ethPrice)
+      await cETH.setExchangeRate(exchangeRate)
+    })
+    it('should return the price in 1e18', async () => {
+      // how much 1e8 cToken worth in USD
+      const cTokenprice = await compoundPricer.getPrice()
+      const expectResult = await underlyingPriceToCtokenPrice(new BigNumber(ethPrice), exchangeRate, weth)
+      assert.equal(cTokenprice.toString(), expectResult.toString())
+      // hard coded answer
+      // 1 cETH = 9.4 USD
+      assert.equal(cTokenprice.toString(), '9409058565621361934')
+    })
+    it('should return the new price after resetting answer in aggregator', async () => {
+      const newPrice = createScaledNumber(500)
+      await chainlinkPricer.setPrice(newPrice)
+      const cTokenPrice = await compoundPricer.getPrice()
+      const expectedResult = await underlyingPriceToCtokenPrice(new BigNumber(newPrice), exchangeRate, weth)
+      assert.equal(cTokenPrice.toString(), expectedResult.toString())
+    })
+    it('should revert if price is lower than 0', async () => {
+      await chainlinkPricer.setPrice('0')
+      await expectRevert(compoundPricer.getPrice(), 'CompoundPricer: underlying price is 0')
+    })
```

Comment:
Can you add a test to see if it works for cUSDC, cDai as well? If cUSDC works, then it tests for the different decimals properly.